syntax = "proto3";

package keyvaluestore;
import "google/protobuf/empty.proto";

service KeyValueStore {
  rpc init(InitRequest) returns (InitResponse);
  rpc shutdown(google.protobuf.Empty) returns (ShutdownResponse);
  rpc get (GetRequest) returns (GetResponse);
  rpc put (PutRequest) returns (PutResponse);
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
  bool key_found = 2;
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
    string old_value = 1;
    bool key_found = 2;
}

message InitRequest {
  string server_name = 1;
}

message InitResponse {
  bool success = 1;
}

message ShutdownResponse {
  bool success = 1;
}
